pandas/io/sql.py

```diff
@@ -2253,6 +2265,10 @@ def drop_table(self, name: str, schema: str | None = None) -> None:
         drop_sql = f"DROP TABLE {_get_valid_sqlite_name(name)}"
         self.execute(drop_sql)
 
+    def trunc_table(self, name: str, schema: str | None = None) -> None:
```

Comment:
I think this method should be deleted, or should explicitly raise a `NotImplementedError` for sqlite. Can you also set up a test called `test_sqlite_truncate_raises` that makes sure that happens? You'll see that pattern in many of the other tests